datafusion-examples/examples/advanced_parquet_index.rs

```diff
@@ -300,8 +300,11 @@ impl IndexTableProvider {
         // In this example, we use the PruningPredicate's literal guarantees to
         // analyze the predicate. In a real system, using
         // `PruningPredicate::prune` would likely be easier to do.
-        let pruning_predicate =
-            PruningPredicate::try_new(Arc::clone(predicate), self.schema())?;
+        let pruning_predicate = PruningPredicate::try_new(
+            Arc::clone(predicate),
+            self.schema(),
+            vec![ColumnOrdering::Unknown; self.schema().fields().len()],
+        )?;
```

Comment:
We could add a new signature to avoid API churn, but I wanted to make it explicit for now to see all of the callsites